src/Microsoft.AspNetCore.Mvc.ApiExplorer/ApiResponseTypeProvider.cs

```diff
@@ -0,0 +1,339 @@
+            {
+                return null;
+            }
+
+            // If we get here, the type should be a user-defined data type or an envelope type
+            // like ActionResult<T>. The mapper service will unwrap envelopes.
+            unwrappedType = _mapper.GetResultDataType(unwrappedType);
+            return unwrappedType;
+        }
+
+        private Type GetRuntimeReturnType(Type declaredReturnType)
+        {
+            // If we get here, then a filter didn't give us an answer, so we need to figure out if we
+            // want to use the declared return type.
+            //
+            // We've already excluded Task, void, and IActionResult at this point.
+            //
+            // If the action might return any object, then assume we don't know anything about it.
+            if (declaredReturnType == typeof(object))
+            {
+                return null;
+            }
+
+            return declaredReturnType;
+        }
+
+        internal static bool IsMatch(MethodInfo methodInfo, MethodInfo conventionMethod)
+        {
+            if (!IsMethodNameMatch(methodInfo.Name, conventionMethod.Name))
+            {
+                return false;
+            }
+
+            var methodParameters = methodInfo.GetParameters();
+            var conventionMethodParameters = conventionMethod.GetParameters();
+            if (conventionMethodParameters.Length != methodParameters.Length)
+            {
+                return false;
+            }
+
+            for (var i = 0; i < conventionMethodParameters.Length; i++)
+            {
+                if (conventionMethodParameters[i].ParameterType.IsGenericParameter)
+                {
+                    // Use TModel as wildcard
+                    continue;
+                }
+                else if (!IsParameterNameMatch(methodParameters[i].Name, conventionMethodParameters[i].Name) ||
+                    !IsParameterTypeMatch(methodParameters[i].ParameterType, conventionMethodParameters[i].ParameterType))
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        internal static bool IsMethodNameMatch(string name, string conventionName)
+        {
+            if (!name.StartsWith(conventionName, StringComparison.Ordinal))
+            {
+                return false;
+            }
+
+            if (name.Length == conventionName.Length)
+            {
+                return true;
+            }
+
+            return char.IsUpper(name[conventionName.Length]);
+        }
+
+        internal static bool IsParameterNameMatch(string name, string conventionName)
+        {
+            // Leading underscores could be used to allow multiple parameter names with the same suffix e.g. GetPersonAddress(int personId, int addressId)
+            // A common convention that allows targeting these category of methods would look like Get(int id, int _id)
```

Comment:
Is there somewhere that you've written down the 'dialect' of authoring a convention?

Reply:
It's listed here - https://github.com/aspnet/Mvc/issues/7802, but I can add more content + examples